src/SignatureHelpProvider.php

```diff
@@ -0,0 +1,111 @@
+<?php
+declare(strict_types = 1);
+
+namespace LanguageServer;
+
+use Microsoft\PhpParser\Node\DelimitedList\ArgumentExpressionList;
+use Microsoft\PhpParser\Node\Expression\CallExpression;
+use Microsoft\PhpParser\Node\Expression\ArgumentExpression;
+use LanguageServer\Index\ReadableIndex;
+use LanguageServer\Protocol\{
+    Range,
+    Position,
+    SignatureHelp,
+    SignatureInformation,
+    ParameterInformation
+};
+
+class SignatureHelpProvider
+{
+    /**
+     * @var DefinitionResolver
+     */
+    private $definitionResolver;
+
+    /**
+     * @var ReadableIndex
+     */
+    private $index;
+
+    /**
+     * @param DefinitionResolver $definitionResolver
+     * @param ReadableIndex      $index
+     */
+    public function __construct(DefinitionResolver $definitionResolver, ReadableIndex $index)
+    {
+        $this->definitionResolver = $definitionResolver;
+        $this->index = $index;
+    }
+
+    /**
+     * Get the short declaration for a callable (class modifiers, function keyword, etc are stripped)
+     *
+     * @param string $declaration
+     * @return string
+     */
+    protected function getShortDeclaration(string $declaration): string
+    {
+        $parts = explode('(', $declaration, 2);
+        $name = array_reverse(explode(' ', trim($parts[0])))[0];
+        return $name . '(' . $parts[1];
```

Comment:
this method looks like it would ignore many edge cases, for example `(` or spaces inside a default string value.
Why attempt to explode a declaration string when we have a parser available? I would add the wanted output to the `Definition` class